viewing_party/party.py

```diff
@@ -1,23 +1,184 @@
 # ------------- WAVE 1 --------------------
 
+from audioop import avg
+from email.policy import default
+from errno import EDEADLK
+from symbol import subscript
+from xml.dom.minidom import Element
+
+
 def create_movie(title, genre, rating):
-    pass
+    #return None for invaild input
+    if title == None or genre == None or rating == None:
+        return None
+    
+    # create dict with given input
+    new_movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating,
+    }
+
+    # return the dict
+    return new_movie 
+
+    # add movie dict to watched list
+def add_to_watched(user_data,movie):
+    user_data["watched"].append(movie)
+    return user_data
+
 
+    # add movie to watchlist
+def add_to_watchlist(user_data,movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+    #  move movie from watchlist to watched
+def watch_movie(user_data,title):
+    watchlist = user_data["watchlist"]
+    watched = user_data["watched"]
+    for movie in watchlist:
+        if movie["title"] == title:
+            watched.append(movie)
+            watchlist.remove(movie)
+                
+    return user_data
+    
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    watched = user_data["watched"]
+    rating_list = []
+    for movie in watched:
+       rating_list.append(movie["rating"])
+    avg = sum(rating_list) / (len(rating_list) or 1)
+    return avg
+
+  
+ 
+     
+
+def get_most_watched_genre(user_data):
+     watched = user_data["watched"]
+     watched_dict = {}
+     for movie in watched:
+        genre = movie["genre"]
+
+        if movie not in watched:
+            return None
+
+        if genre not in watched_dict:
+            watched_dict[genre] = 1 
+        else: 
+            watched_dict[genre] += 1 
+               
+        most_watched = max(watched_dict)
+        return most_watched
 
+        
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_friends_movies(user_data):
+    friends_watched_list = []
+    friends_watched_list_of_dicts = user_data["friends"]
+
+    for index in range(len(friends_watched_list_of_dicts)):
+        for index_2 in range(len(friends_watched_list_of_dicts[index]["watched"])):
+            friends_watched_list.append(friends_watched_list_of_dicts[index]["watched"][index_2])
+
+    return friends_watched_list
+
+
+def get_unique_watched(user_data):
+    unique_movies = []
+    user_watched_list = user_data["watched"]
+    friends_watched_list = get_friends_movies(user_data)
+
+    for index in range(len(user_watched_list)):

```

Comment:
I noticed you used your custom function to grab the friends watched list, you could also access the list by looping:
```
    for friend in user_data["friends"]:
        for movie in friend["watched"]:
``` 

Additionally, instead of using range to loop over the length of the user watchlist and accessing `user_watched_list[index]` you could loop over watchlist and for each watch you could access and compare the title. Something like this:
```
    for watched in user_data["watched"]:
        if not watched in friend["watched"]:
```